keps/sig-cloud-provider/2699-add-webhook-hosting-to-ccm/README.md

```diff
@@ -0,0 +1,541 @@
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP will detail enhancing the CCM framework to support cloud provider
+specific webhooks. The intent is to make it easy to either generate a binary
+or enhance the existing CCM binary to host such webhooks. We also intend to
+allow for easily linking in "standard" webhooks needed by other SIGs which
+need to be customized for particular cloud providers.
+
+## Motivation
+
+The Cloud (Provider) Controller Manager (CCM) is the binary into which the Cloud
+Provider places all the controllers needed to make a Kubernetes cluster work
+correctly on their Cloud. There are also occasions when it makes sense for a Cloud
+Provider to want these customizations to be applied inline rather than 
+asynchronously after a change has already been applied. 
+
+Our initial example of this is from SIG Storage. These would like the functionality 
+from PVL admission controller (https://github.com/kubernetes/kubernetes/issues/52617).
+This needs to be completed for cloud provider extraction to complete. Several
+Cloud Providers have indicated that this should be done inline, especially as the
+existing deprecated solution is an inline solution.
+
+### Goals
+
+Our imediate goal is to allow in tree Cloud Providers are able to stop using the 
+existing PVL admission controller and do so using the framework. However we want to
+build a framework which wil be usable by similar solutions to problems. This KEP is
+about the framework needed to support the PVL webhook and not the webhook itself.
+
+### Non-Goals
+
+We are not intending to create a [general admission webhook solution](https://book.kubebuilder.io/cronjob-tutorial/webhook-implementation.html). This is just
+intended to host Cloud Provider specific webhooks as part of the Control Plane.
+
+## Proposal
+
+We will start by adding extension hooks which can be registered in the 
+cmd/cloud-controller-manager/main.go. This would be similar to the mechanism
+we already use to register new controllers. The existing sample shows this with
+a sample of registering the nodeipamcontroller which is not a normally installed
+controller in the cloud controller manager. In a similar way we will have a 
+sample of integrating a PVL mutating webhook into the sample CCM. We will also 
+have the system automatically detect if there are both controllers and webhooks
+registered in the binary. If both are registered it will automatically add 
+command line flags allowing webhooks and controller to be disabled. The 
```

Comment:
I assume this would be two separate flags right?
```suggestion
registered in the binary. If both are registered it will automatically add 
command line flags allowing webhooks and/or controllers to be disabled. The 
```

Reply:
Yes, I talk about both the controllers flag and the webhooks flag in the next few sentences. However its probably better to be explicit, so I've added a sentence to make that explicit.